library/iFixit/Matryoshka/Memcached.php

```diff
@@ -60,16 +89,38 @@ public function decrement($key, $amount = 1, $expiration = 0) {
        $expiration);
    }
 
+   /**
+    * @template T
+    * @return T|self::MISS
+    */
    public function get($key) {
+      /** @var T|false */
       $value = $this->memcached->get($key);
+      $result = $this->memcached->getResultCode();
 
-      if ($this->memcached->getResultCode() === \Memcached::RES_NOTFOUND) {
+      if ($this->isMiss($result)) {
          return self::MISS;
       }
 
+      $error = $this->getError($result);
+      if ($error) {
+         throw $error;
+      }
+
+      /** @var T */
       return $value;
    }
 
+   private function isMiss(int $resultCode) {
+      return $resultCode === \Memcached::RES_NOTFOUND;
+   }
+
+   private function getError(int $resultCode): ?\MemcachedException {
+      $hasKey = array_key_exists($resultCode, self::ERRORS);
+      $errorMessage = $hasKey ? self::ERRORS[$resultCode] : null;
```

Comment:
Nah, lets have a sane default message here. Empty exception strings are _weird_...

```suggestion
      $errorMessage = $hasKey
       ? self::ERRORS[$resultCode]
       : "An unknown error occured: {$resultCode}";
```

What do you think of that? I'm not tied to that exact message, but it should be something helpful...

Reply:
We run mcrouter in prod, which might have it's own special error numbers...

Reply:
Oh right now I also pass success into this function - it's just an error code. I have covered the width of errors in the array. I don't mind making found/not found early exists, then a default error message. Opinions?

Reply:
Huh? I'm just talking about the default exception message if we don't know about a specific error number.

Reply:
If we are going to treat an unknown result code as an error, we'll have to early exit if `resultCode == success`.